server/graphql/programs/src/queries/program_event.rs

```diff
@@ -44,7 +42,62 @@ pub struct ProgramEventSortInput {
 pub fn program_events(
     ctx: &Context<'_>,
     store_id: String,
-    patient_id: String,
+    page: Option<PaginationInput>,
```

Comment:
Ideally the API shouldn't change in breaking way. Currently, the user would get an error while trying to do the report query. There are scripts in the report repository to print a report using one command. Maybe we should have a mechanism/script to run all reports in sequence to be able to test them...

Reply:
> Ideally the API shouldn't change in breaking way.

Do you think this is possible to guarantee (for the rest of omSupply life) ? 

> Maybe we should have a mechanism/script to run all reports in sequence to be able to test them

How would this work, since configurations are on central server, which could have different remote servers on different versions. 

Btw I think it's important to answer this question and build a mechanism for this asap. If breaking API changes are not allowed before we figure this out, then we need to communicate this to the whole team

Reply:
> > Ideally the API shouldn't change in breaking way.
> 
> Do you think this is possible to guarantee (for the rest of omSupply life) ?

yes, for example, by either introducing new endpoints like `myEndpointV2` if breaking changes are required. It gets trickier if we really want to retire a certain functionality... There seems to be some schema checkers to check for breaking changes but not sure how actively they are developed or how good they are...

> > Maybe we should have a mechanism/script to run all reports in sequence to be able to test them
> 
> How would this work, since configurations are on central server, which could have different remote servers on different versions.

Yes good point. Same problem exist for automated UI testing. We could use the DB json snapshots for this? could become quite complex though, e.g. to correctly update the DB snapshot...

> 
> Btw I think it's important to answer this question and build a mechanism for this asap. If breaking API changes are not allowed before we figure this out, then we need to communicate this to the whole team

Kind of assumed we are doing it even though I never really cared about it much since we always update backend and UI at the same time (thus an API change doesn't break the UI). However, the reports are a good point and we should at least make sure we don't break these endpoints!